### PR TITLE
Delete multi-files mode of XIIDM import/export

### PR DESCRIPTION
**Pull request by MioRtia**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?**
No

**What kind of change does this PR introduce?**
Delete multi-files mode of XIIDM import/export: its usefulness is not significant enough regarding the performance regression it introduces.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [x] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

### CODE REVIEW

File: commons/src/main/java/com/powsybl/commons/xml/XmlUtil.java [hunk 1]
```diff
@@ -173,4 +179,29 @@ public static float readOptionalFloatAttribute(XMLStreamReader reader, String at
...
+
+    public static XMLStreamWriter writeStartAttributes(OutputStream os, String prefix, String namespaceUri, String rootName, String indentString, boolean indent) throws XMLStreamException {
+        XMLStreamWriter writer;
+        writer = createXmlStreamWriter(indentString, indent, os);
+        writer.writeStartDocument(StandardCharsets.UTF_8.toString(), "1.0");
+        writer.setPrefix(prefix, namespaceUri);
+        writer.writeStartElement(namespaceUri, rootName);
+        writer.writeNamespace(prefix, namespaceUri);
+        return writer;
+    }
```
Comment (mathbagu) [inline commons/src/main/java/com/powsybl/commons/xml/XmlUtil.java]: Is it a good idea to mutualize this?

File: commons/src/main/java/com/powsybl/commons/xml/XmlUtil.java [hunk 2]
```diff
@@ -173,4 +179,29 @@ public static float readOptionalFloatAttribute(XMLStreamReader reader, String at
...
+            indentingWriter.setIndent(indentString);
+            writer = indentingWriter;
+        }
+        return writer;
+    }
+
+    public static void writeEndElement(XMLStreamWriter writer) throws XMLStreamException {
+        writer.writeEndElement();
+        writer.writeEndDocument();
+    }
```
Comment (mathbagu) [inline commons/src/main/java/com/powsybl/commons/xml/XmlUtil.java]: Same question? The name of this method is not OK. As a developer, I don't know it will end my XML document.

File: iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/IidmImportExportMode.java [hunk 3]
```diff
@@ -8,12 +8,14 @@
...
+ * @deprecated Not used anymore
  */
-
+@Deprecated
 public enum IidmImportExportMode {
     ONE_SEPARATED_FILE_PER_EXTENSION_TYPE,
-    UNIQUE_FILE,
-    EXTENSIONS_IN_ONE_SEPARATED_FILE
+    EXTENSIONS_IN_ONE_SEPARATED_FILE,
+    UNIQUE_FILE
```
Comment (mathbagu) [inline iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/IidmImportExportMode.java]: You should revert this change: if bad people use the order of this enum, they can consider it as a regression. Keep only the `@Deprecated`

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java [hunk 4]
```diff
@@ -436,62 +326,23 @@ public static Anonymizer write(Network network, ExportOptions options, DataSourc
...
 
-    public static Anonymizer writeAndValidate(Network n, ExportOptions options, Path xmlFile) throws IOException {
+    public static Anonymizer writeAndValidate(Network n, ExportOptions options, Path xmlFile) {
         Anonymizer anonymizer = write(n, options, xmlFile);
-        if (options.getMode() == IidmImportExportMode.UNIQUE_FILE) {
-            validateWithExtensions(xmlFile);
-            return anonymizer;
-        }
-        validate(xmlFile, options.getMode());
+        validateWithExtensions(xmlFile);
```
Comment (mathbagu) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java]: Is there a validateWithoutExtension?

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java [hunk 5]
```diff
@@ -618,140 +467,32 @@ public static Network read(ReadOnlyDataSource dataSource, NetworkFactory network
...
-            return read(xmlFile);
-        }
-        validate(xmlFile, options.getMode());
-        return read(xmlFile, options);
+    /**
+     * @deprecated Use {@link #read(Path)} instead.
+     */
+    @Deprecated
+    public static Network read(Path xmlFile, ImportOptions options) {
+        return read(xmlFile);
```
Comment (mathbagu) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java]: Is it a good idea to forget the Import/export Options? Are you sure we should not keep both? As the ImportOptions is not deprecated, why this method is?

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java [hunk 6]
```diff
@@ -618,140 +467,32 @@ public static Network read(ReadOnlyDataSource dataSource, NetworkFactory network
...
-                    LOGGER.warn(String.format("the %s file is not found ", fileName));
-                }
-            }
-        }
+    /**
+     * @deprecated Use {@link #validateAndRead(Path)} instead.
+     */
+    @Deprecated
+    public static Network validateAndRead(Path xmlFile, ImportOptions options) {
+        return validateAndRead(xmlFile);
```
Comment (mathbagu) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java]: Same question

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java [hunk 7]
```diff
@@ -618,140 +467,32 @@ public static Network read(ReadOnlyDataSource dataSource, NetworkFactory network
...
-            });
-
-            checkExtensionsNotFound(context, extensionNamesNotFound);
-
-            return network;
-        } catch (XMLStreamException e) {
-            throw new UncheckedXmlStreamException(e);
-        }
+    public static Network validateAndRead(Path xmlFile) {
+        validateWithExtensions(xmlFile);
```
Comment (mathbagu) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java]: I think you should rename this method `validate`

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java [hunk 8]
```diff
@@ -149,4 +138,16 @@ private void addExtensionsVersions(Properties parameters, ExportOptions options)
...
+                .setOnlyMainCc(ConversionParameters.readBooleanParameter(getFormat(), parameters, ONLY_MAIN_CC_PARAMETER, defaultValueConfig))
+                .setAnonymized(ConversionParameters.readBooleanParameter(getFormat(), parameters, ANONYMISED_PARAMETER, defaultValueConfig))
+                .setTopologyLevel(TopologyLevel.valueOf(ConversionParameters.readStringParameter(getFormat(), parameters, TOPOLOGY_LEVEL_PARAMETER, defaultValueConfig)))
+                .setThrowExceptionIfExtensionNotFound(ConversionParameters.readBooleanParameter(getFormat(), parameters, THROW_EXCEPTION_IF_EXTENSION_NOT_FOUND_PARAMETER, defaultValueConfig))
+                .setExtensions(ConversionParameters.readStringListParameter(getFormat(), parameters, EXTENSIONS_LIST_PARAMETER, defaultValueConfig) != null ? new HashSet<>(ConversionParameters.readStringListParameter(getFormat(), parameters, EXTENSIONS_LIST_PARAMETER, defaultValueConfig)) : null)
+                .setVersion(ConversionParameters.readStringParameter(getFormat(), parameters, VERSION_PARAMETER, defaultValueConfig));
+        addExtensionsVersions(parameters, options);
+    }
```
Comment (mathbagu) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java]: Do you think it's better with a method? Why don't you finish the job here:
```
private ExportOptions createExportOptions(Properties parameters) {
    ...
}
```

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLImporter.java [hunk 9]
```diff
@@ -195,5 +187,12 @@ public Network importData(ReadOnlyDataSource dataSource, NetworkFactory networkF
         }
         return network;
     }
+
+    protected void buildImportOptions(Properties parameters, ImportOptions options) {
+        options
+                .setThrowExceptionIfExtensionNotFound(ConversionParameters.readBooleanParameter(getFormat(), parameters, THROW_EXCEPTION_IF_EXTENSION_NOT_FOUND_PARAMETER, defaultValueConfig))
+                .setExtensions(ConversionParameters.readStringListParameter(getFormat(), parameters, EXTENSIONS_LIST_PARAMETER, defaultValueConfig) != null ? new HashSet<>(ConversionParameters.readStringListParameter(getFormat(), parameters, EXTENSIONS_LIST_PARAMETER, defaultValueConfig)) : null);
+
+    }
```
Comment (mathbagu) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLImporter.java]: Same remark

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java (hunk 4 above)
Comment (MioRtia) [reply iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java]: There was but as it is not used anymore, I will rename this one as you suggested
